support-workers/src/main/scala/com/gu/stripe/StripeService.scala

```diff
@@ -19,36 +20,41 @@ import com.gu.support.zuora.api.{
 import io.circe.{Decoder, Json}
 import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
 import com.gu.support.workers.exceptions.{RetryException, RetryLimited, RetryNone, RetryUnlimited}
+import com.typesafe.scalalogging.StrictLogging
 import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
 import io.circe.syntax._
 import io.circe.{Decoder, Encoder, Json}
+
 import scala.concurrent.{ExecutionContext, Future}
 import scala.reflect.ClassTag
 
-class StripeService(val config: StripeConfig, client: FutureHttpClient, baseUrl: String = "https://api.stripe.com/v1")(
-    implicit ec: ExecutionContext,
-) extends WebServiceHelper[StripeError] {
+class StripeService(val config: StripeConfig, client: FutureHttpClient, baseUrl: String = "https://api.stripe.com/v1")
+    extends WebServiceHelper[StripeError]
+    with StrictLogging {
 
   // Stripe URL is the same in all environments
   val wsUrl = baseUrl
   val httpClient: FutureHttpClient = client
 
-  def withCurrency(currency: Currency): StripeServiceForCurrency = new StripeServiceForCurrency(this, currency)
-
-}
-
-object StripeServiceForCurrency {
-
-  def chargeGateway(currency: Currency): PaymentGateway =
-    if (currency == AUD) StripeGatewayAUD else StripeGatewayDefault
+  def withCurrency(currency: Currency): StripeServiceForCurrency = {
+    logger.warn(s"DEPRECATED FALLBACK - using currency $currency to determine stripe gateway")
```

Comment:
this is here so I can just search the support-workers log quickly and find out all/any checkouts that are not yet sending the public key in the payload.